service/key/key.go

```diff
@@ -1,17 +1,34 @@
 package key
 
 import (
+	"fmt"
+
 	"github.com/giantswarm/azuretpr"
+	"github.com/giantswarm/certificatetpr"
 )
 
+// ClusterCustomer returns the customer ID for this cluster.
 func ClusterCustomer(customObject azuretpr.CustomObject) string {
 	return customObject.Spec.Cluster.Customer.ID
 }
 
+// ClusterID returns the unique ID for this cluster.
 func ClusterID(customObject azuretpr.CustomObject) string {
 	return customObject.Spec.Cluster.Cluster.ID
 }
 
+// KeyVaultName returns the Azure Key Vault name for this cluster.
+func KeyVaultName(customObject azuretpr.CustomObject) string {
+	return customObject.Spec.Azure.KeyVault.VaultName
+}
+
+// Location returns the physical location where the Resource Group is deployed.
 func Location(customObject azuretpr.CustomObject) string {
 	return customObject.Spec.Azure.Location
 }
+
+// SecretName returns the name of the Key Vault secret for this certificate
+// asset.
```

Comment:
A unique secret name is generated within the vault for the certs files. They can't be the file name unfortunately.

> An object-name is a user provided name for and must be unique within a Key Vault. The name must be a string 1-127 characters in length containing only 0-9, a-z, A-Z, and -.